### PR TITLE
Update minimum wage rates for October 2016

### PR DESCRIPTION
**Pull request by robinjam**
[Trello card](https://trello.com/c/7BwRqtZ3)

## Description

New rates (in effect from 1st October 2016 onwards):
* Under 18: £4.00
* 18-20: £5.55
* 21-24: £6.95
* 25+: £7.20 (no change)
* Apprentices: £3.40
* Accommodation: £6.00

This change affects the following smart answers:
* Minimum wage calculator for workers (am-i-getting-minimum-wage)
* Minimum wage calculator for employers (minimum-wage-calculator-employers)

## Expected changes - before 1st October 2016

If this is deployed before the 1st of October 2016, there should be no user-visible changes to either of these pages:

* https://www.gov.uk/am-i-getting-minimum-wage/y/current_payment/not_an_apprentice/20/30/160.0/850.0/0.0/no
* https://www.gov.uk/minimum-wage-calculator-employers/y/current_payment/not_an_apprentice/20/30/160.0/850.0/0.0/no

## Expected changes - after 1st October 2016

After midnight on the 1st of October, the following user-visible changes should be present:

#### Before

https://www.gov.uk/am-i-getting-minimum-wage/y/current_payment/not_an_apprentice/20/30/160.0/850.0/0.0/no

screen shot 2016-09-22 at 15 39 51

#### After

https://smart-answers-pr-2741.herokuapp.com/am-i-getting-minimum-wage/y/current_payment/not_an_apprentice/20/30/160.0/850.0/0.0/no

screen shot 2016-09-22 at 15 39 54

#### Before

https://www.gov.uk/minimum-wage-calculator-employers/y/current_payment/not_an_apprentice/20/30/160.0/850.0/0.0/no

screen shot 2016-09-22 at 15 41 28

#### After

https://smart-answers-pr-2741.herokuapp.com/minimum-wage-calculator-employers/y/current_payment/not_an_apprentice/20/30/160.0/850.0/0.0/no

screen shot 2016-09-22 at 15 41 30

### CODE REVIEW

**Comment by leenagupte** [comment]
@robinjam I think it's worth adding a few unit tests, with the date set to Oct 1st 2016, that checks the new minimum wage rates.

File: lib/smart_answer/calculators/minimum_wage_calculator.rb [hunk 1]
```diff
@@ -208,5 +208,13 @@ def charged_accomodation_adjustment(charge, number_of_nights)
     def self.historical_minimum_wage_data
       @historical_minimum_wage_data ||= YAML.load_file(Rails.root.join("lib/data/minimum_wage_data.yml"))[:minimum_wage_data]
     end
+
+    def date_from_environment_variable
```
Comment (leenagupte) [inline lib/smart_answer/calculators/minimum_wage_calculator.rb]: @robinjam This function already exists in `smart-answers/lib/smart_answer/calculators/rates_query.rb`. Why have you re-implemented it here?

Comment (robinjam) [inline lib/smart_answer/calculators/minimum_wage_calculator.rb]: @leenagupte `MinimumWageCalculator` doesn't use `RatesQuery` (in fact, it pre-dates it by about 3 years).

They do different things as well: this function returns `Date.now` if `RATES_QUERY_DATE` is unset, whereas the implementation in `RatesQuery` returns `nil` in that situation.

Comment (leenagupte) [inline lib/smart_answer/calculators/minimum_wage_calculator.rb]: @robinjam I would update the minimum wage calculator to get its rates from RatesQuery, just like the other calculators.

This will make life simpler because there will be less duplication (rates files are only read in one place) and we wouldn't need to re-implement the date_from_environment_variable function.

Can you create an initial commit that changes the minimum wage calculators to use RatesQuery, and then make the new rates changes after that commit?

The married-couples-allowance is a good example of how the RatesQuery is used.

Comment (robinjam) [inline lib/smart_answer/calculators/minimum_wage_calculator.rb]: @leenagupte Since someone else is going to be taking over Smart Answers next week (and given it's 4:30PM on Friday now), I think that should maybe be a new card. What do you reckon?

I can walk through the required changes with the next person during the handover meeting (10am Monday).

Comment (leenagupte) [inline lib/smart_answer/calculators/minimum_wage_calculator.rb]: @robinjam, @pmanrubia says he is ready and eager to take over this story (and make the changes) on Monday.

Comment (pmanrubia) [inline lib/smart_answer/calculators/minimum_wage_calculator.rb]: @leenagupte your deserve it => :trollface:

**Comment by pmanrubia** [comment]
Superseded by #2747
